From 76c9855c7c854ebeb2f85610d863869f03e30ea1 Mon Sep 17 00:00:00 2001
From: Aurelian Zanoschi <Aurelian.Zanoschi@freescale.com>
Date: Fri, 19 Apr 2013 17:50:59 +0300
Subject: [PATCH 399/547] dpa_stats: Add support for UNICAST class counter

Enabled the UNICAST IN/OUT class counter support into DPA Stats
component by removing the condition that returned an error in case
of UNICAST class counter was selected.

Signed-off-by: Aurelian Zanoschi <Aurelian.Zanoschi@freescale.com>
Change-Id: Ib36048bc2a4a1c1d3d4f9fe511c35d44a4fbcb62
Reviewed-on: http://git.am.freescale.net:8181/2240
Reviewed-by: Floarea Anca Jeanina-B12569 <anca.floarea@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
[Original patch taken from QorIQ-SDK-V1.4-SOURCE-20130625-yocto.iso]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 drivers/staging/fsl_dpa_offload/dpa_stats.c | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/dpa_stats.c b/drivers/staging/fsl_dpa_offload/dpa_stats.c
index e53227b..13f550c 100644
--- a/drivers/staging/fsl_dpa_offload/dpa_stats.c
+++ b/drivers/staging/fsl_dpa_offload/dpa_stats.c
@@ -1463,16 +1463,6 @@ static int set_cls_cnt_eth_cb(struct dpa_stats_cnt_cb *cnt_cb,
 		cnt_sel -= 1;
 
 	for (i = 0; i < params->class_members; i++) {
-		/* 1G Port, DTSEC controller */
-		if (params->eth_params.src[i].eth_id <
-				DPA_STATS_ETH_10G_PORT0) {
-			if (cnt_sel & DPA_STATS_CNT_ETH_IN_UNICAST_PKTS ||
-			    cnt_sel & DPA_STATS_CNT_ETH_OUT_UNICAST_PKTS) {
-				pr_err("Invalid Ethernet counter selection\n");
-				return -EINVAL;
-			}
-		}
-
 		/* Get FM MAC handle */
 		err = get_fm_mac(params->eth_params.src[i], &fm_mac);
 		if (err != 0) {
-- 
1.8.4.93.g57e4c17

